src/panels/experimental-ui/hui-entity-card.js

```diff
@@ -0,0 +1,66 @@
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import computeDomain from '../../common/entity/compute_domain.js';
+import { DOMAINS_WITH_CARD } from '../../common/const.js';
+
+import '../../cards/ha-camera-card.js';
+import '../../cards/ha-history_graph-card.js';
+import '../../cards/ha-media_player-card.js';
+import '../../cards/ha-persistent_notification-card.js';
+import '../../cards/ha-plant-card.js';
+import '../../cards/ha-weather-card.js';
+
+class HuiEntityCard extends PolymerElement {
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: '_hassChanged',
+      },
+      config: {
+        type: Object,
+        observer: '_configChanged',
+      }
+    };
+  }
+
+  getCardSize() {
+    return this._entityId ? DOMAINS_WITH_CARD[computeDomain(this._entityId)] : 3;
+  }
+
+  _configChanged(config) {
+    this._entityId = null;
+    if (this.childNodes.length) {
+      this.removeChild(this.childNodes[0]);
+    }
+
+    const entityId = config && config.entity;
+    if (entityId) {
+      const domain = computeDomain(entityId);
+      if (Object.keys(DOMAINS_WITH_CARD).includes(domain)) {
```

Comment:
I think that this is the wrong approach. We're adding magic again and we're trying to get rid of all magic.

Instead, if people want to have full cards, we need to add them as a card type that they can specify in their config. It should not be triggered by a list that we maintain.

When adding a new card type, add a new file that starts with `hui-`. It can still use the old card element internally. We should not add `getCardSize` to the old card elements. I want to keep that separated.

Reply:
Oh, I see now that you did this already, oops 😱 

I still think that we should consider 1 file per card. That way we can start adding config and validation per file. Makes it easier to do a UI in the future.

Reply:
so you suggest this as config?
```
- type: media_player
  entity: media_player.my_chromecast
```

Reply:
... in this PR the user can use the internal card with
```yaml
- type: entity
  entity: media_player.mini
```
 or
```yaml
- type: custom:my-custom-sensor-card
  entity: sensor.test
```

bot as big card

Reply:
I agree that we should split it into multiple files, but if we define the domain as type again it's redundant

Reply:
I think that the type should incorporate what the card does. So things like:

 - media-control
 - camera-preview